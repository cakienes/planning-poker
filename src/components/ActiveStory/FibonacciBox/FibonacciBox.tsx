import React from 'react';
import IFibonacciBoxProps from './interface/IFibonacciBoxProps';

export class FibonacciBox extends React.Component<IFibonacciBoxProps, any> {
    render() {
        const { fibonacciNumber, storyPoint } = this.props;
        return (
            <li
                className={`col-3 ${storyPoint === fibonacciNumber ? 'selected' : undefined}`}
                onClick={this.onClickFibonacci}
            >
                <span>{fibonacciNumber || '?'}</span>
            </li>
        );
    }

    onClickFibonacci = (): void => {
        const { fibonacciNumber } = this.props;
        this.props.giveStoryPoint(fibonacciNumber);
    };
}

export default FibonacciBox;
