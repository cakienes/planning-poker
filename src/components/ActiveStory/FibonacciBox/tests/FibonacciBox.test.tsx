import React from 'react';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import FibonacciBox from '../FibonacciBox';

describe('FibonacciBox.tsx', () => {
    let renderer: ShallowRenderer;

    const giveStoryPointMock = jest.fn();
    let fibonacciBoxProps: any = {};

    beforeEach(() => {
        renderer = createRenderer();

        fibonacciBoxProps = {
            giveStoryPoint: giveStoryPointMock,
            fibonacciNumber: 1,
            storyPoint: 1,
        };
    });

    it('renders correctly FibonacciBox', async () => {
        expect(renderer.render(<FibonacciBox {...fibonacciBoxProps} />)).toMatchSnapshot();
    });

    it('renders correctly FibonacciBox while fibonacciNumber is 0', async () => {
        fibonacciBoxProps = {
            ...fibonacciBoxProps,
            fibonacciNumber: 0,
            storyPoint: 1,
        };
        renderer.render(<FibonacciBox {...fibonacciBoxProps} />);
        const instance: any = renderer.getMountedInstance();

        expect(instance.render()).toMatchSnapshot();
    });

    it('onClickFibonacci of FibonacciBox', async () => {
        renderer.render(<FibonacciBox {...fibonacciBoxProps} />);
        const instance: any = renderer.getMountedInstance();

        instance.onClickFibonacci();

        expect(giveStoryPointMock).toHaveBeenCalledWith(1);
    });
});
