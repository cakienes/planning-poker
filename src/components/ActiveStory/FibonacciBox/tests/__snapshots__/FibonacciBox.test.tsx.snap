// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FibonacciBox.tsx renders correctly FibonacciBox 1`] = `
<li
  className="col-3 selected"
  onClick={[Function]}
>
  <span>
    1
  </span>
</li>
`;

exports[`FibonacciBox.tsx renders correctly FibonacciBox while fibonacciNumber is 0 1`] = `
<li
  className="col-3 undefined"
  onClick={[Function]}
>
  <span>
    ?
  </span>
</li>
`;
